### ISSUE TITLE
`class extends` doesn't work with braced body

### ISSUE THREAD

**Issue by esthedebeste** [DanielXMoore/Civet#543]
```ts
export class CustomError extends Error {
 constructor(message: string) {
  super(message) 
  this.name = "CustomError"
 }
}
```
Compiles to a function call on the `Error` constructor;
```ts
export class CustomError extends Error({
  constructor(message: string) {
    super(message);
    this.name = "CustomError";
  },
}) {}
```

**Comment by edemaine** [comment]
`class X extends Y {}` is a minimal reproduction.  The issue is that `Y {}` looks like an implicit function call, with an object literal.

We had a similar issue with `if`. So far the decision with `if` is that, if you want to brace the body of an `if`, then you need to wrap the condition in parentheses (full JS compatibility, but no mixing and matching). I still find this confusing, so when we fix `class ... extends` (which seems important), perhaps we could apply the same fix to `if`.

It seems like we need another flag here, where object literals should not trigger implicit function calls. We have a similar flag (already turned on) that disables "indented application" (implicit function calls where the argument is an indented implicit object litearl), so it seems to fit. If you see another fix, let me know; otherwise, I'll take a crack at this.

**Comment by edemaine** [comment]
Fixed in 0.6.13!
